frontend/packages/console-metal3/src/components/host.tsx

```diff
@@ -0,0 +1,150 @@
+import { WithResources } from '@console/shared';
+
+import MachineCell from './machine-cell';
+
+const nameColumnClasses = 'col-lg-2 col-md-4 col-sm-6 col-xs-6';
+const statusColumnClasses = 'col-lg-2 col-md-4 hidden-sm hidden-xs';
+const machineColumnClasses = 'col-lg-3 visible-lg';
+const roleColumnClasses = 'col-lg-2 visible-lg';
+const addressColumnClasses = 'col-lg-2 visible-lg';
+
+const HostHeader = props => (
+  <ListHeader>
+    <ColHead {...props} className={nameColumnClasses} sortField="metadata.name">
+      Name
+    </ColHead>
+    <ColHead {...props} className={statusColumnClasses}>
+      Status
+    </ColHead>
+    <ColHead {...props} className={machineColumnClasses}>
+      Machine
+    </ColHead>
+    <ColHead {...props} className={roleColumnClasses}>
+      Role
+    </ColHead>
+    <ColHead
+      {...props}
+      className={addressColumnClasses}
+      sortField="spec.bmc.address"
+    >
+      Management Address
+    </ColHead>
+  </ListHeader>
+);
+
+const HostRow = ({ obj: host }) => {
+  const name = getName(host);
+  const namespace = getNamespace(host);
+  const uid = getUid(host);
+  const machineName = getHostMachineName(host);
+  const address = getHostBmcAddress(host);
+
+  const machineResource = {
+    kind: referenceForModel(MachineModel),
+    name: machineName,
+    namespaced: true,
+    namespace,
+    isList: false,
+    prop: 'machine'
+  };
+
+  const hostResourceMap = {
+    machine: {
+      resource: machineResource
+    },
+    nodes: {
+      resource: getResource(NodeModel, { namespaced: false })
+    }
+  };
+
+  return (
+    <ResourceRow obj={host}>
+      <div className={nameColumnClasses}>
+        <ResourceLink
+          kind={BaremetalHostModel.kind}
+          name={name}
+          namespace={namespace}
+          title={uid}
+        />
+      </div>
+      <div className={statusColumnClasses}>
+        <WithResources resourceMap={machineName ? hostResourceMap : {}}>
+          <BaremetalHostStatus host={host} />
+        </WithResources>
+      </div>
+      <div className={machineColumnClasses}>
+        <MachineCell host={host} />
+      </div>
+      <div className={roleColumnClasses}>
+        <WithResources resourceMap={machineName ? hostResourceMap : {}}>
+          <BaremetalHostRole />
+        </WithResources>
+      </div>
+      <div className={addressColumnClasses}>{address}</div>
+    </ResourceRow>
+  );
+};
+
+const HostList = props => <List {...props} Header={HostHeader} Row={HostRow} />;
+
+const filters = [
+  {
+    type: 'baremetalhost-status',
+    selected: ['online', 'offline'],
+    reducer: getSimpleHostStatus,
+    items: [
+      { id: 'online', title: 'online' },
+      { id: 'offline', title: 'offline' }
+    ]
+  }
+];
+
+/* eslint-disable no-unused-vars, no-undef */
+export type BaremetalHostsPageProps = {
+  namespace: string;
+};
+
+export class BaremetalHostsPage extends React.Component<
```

Comment:
Use a functional component instead of a class

Reply:
Done